feat: modernize UI with MUI and ODE tokens

Bahati308 opened the pull request:
closes #9

Bahati308 committed:
feat: modernize UI with MUI and ODE tokens

- Implement Material Design 3 theme with ODE token integration
- Modernize buttons: pill-shaped, proper elevation, 48dp touch targets
- Update inputs: 4px radius, improved focus states, 56dp min height
- Apply ODE brand colors (green primary, gold secondary)
- Centralize styling in theme for consistent application
- Ensure accessibility: WCAG contrast, proper focus, touch targets.

IamLRBA reviewed (commented):
Nice going @Bahati308, this is actually one of the deliverables of the unified design sytem for ODE and comes at a perfect time. The theme integration, accessibility improvements (48dp touch targets), and component updates all look good.

**Just One Issue**: I think the local `tokens.ts` file you created duplicates token values. If `@ode/tokens` is already in `package.json` of `formulus-formplayer, we should import directly from the original tokens package instead.

**Suggested fix:**
Maybe we can replace the local `tokens.ts` file with direct imports from the package:

In **`theme.ts`**, change from:
**`import { tokens } from './tokens';`**
To:
**`import tokens from '@ode/tokens/dist/js/tokens';`**

This keeps tokens in sync automatically from the original source and avoids duplication. Create React App should handle imports from `node_modules` after `npm install`.

Lastly, we might also consider adding the CSS import to `index.css` for CSS variable support:
`@import '@ode/tokens/dist/css/tokens.css';`

Thanks again for the improvements mate!

IamLRBA commented:
I really love this initiative you have put in this!

Bahati308 commented:
> Nice going @Bahati308, this is actually one of the deliverables of the unified design sytem for ODE and comes at a perfect time. The theme integration, accessibility improvements (48dp touch targets), and component updates all look good.
> 
> **Just One Issue**: I think the local `tokens.ts` file you created duplicates token values. If `@ode/tokens` is already in `package.json` of `formulus-formplayer, we should import directly from the original tokens package instead.
> 
> **Suggested fix:** Maybe we can replace the local `tokens.ts` file with direct imports from the package:
> 
> In **`theme.ts`**, change from: **`import { tokens } from './tokens';`** To: **`import tokens from '@ode/tokens/dist/js/tokens';`**
> 
> This keeps tokens in sync automatically from the original source and avoids duplication. Create React App should handle imports from `node_modules` after `npm install`.
> 
> Lastly, we might also consider adding the CSS import to `index.css` for CSS variable support: `@import '@ode/tokens/dist/css/tokens.css';`
> 
> Thanks again for the improvements mate!

thanks for the insight @IamLRBA , lemme improve accordingly.

Bahati308 committed:
Merge branch 'dev' into Fix-Issue#9

Bahati308 committed:
Merge branch 'dev' into Fix-Issue#9

IamLRBA committed:
Add local dependency for @ode/tokens

IamLRBA committed:
Change tokens import to use tokens-adapter

IamLRBA committed:
Delete formulus-formplayer/src/tokens.ts

IamLRBA committed:
Add tokens adapter for theme.ts compatibility

This file adapts the @ode/tokens JSON export to the expected structure for theme.ts, transforming font properties to typography and parsing pixel values.

IamLRBA commented:
@Bahati308, based on the [suggestion](https://github.com/OpenDataEnsemble/ode/pull/61#pullrequestreview-3570702702), I made some changes please have a look.

## Summary

I fixed the token duplication issue by replacing the local `tokens.ts` file with direct imports from the `@ode/tokens` package. This ensures design tokens stay in sync automatically with the centralized design system.

## Changes Made

- **Removed duplicate `tokens.ts`** - Eliminated 170 lines of token definitions  
- **Added `@ode/tokens` dependency** - Now using the official design tokens package  
- **Created `tokens-adapter.ts`** - Bridges structure differences between the package export and theme requirements  
- **Updated `theme.ts`** - Changed import to use the new adapter instead of local tokens

## Setup Steps

After pulling these changes, run the following to update `package-lock.json` and install dependencies:

```bash
# Build the tokens package first
cd packages/tokens
npm install
npm run build

# Then install dependencies in formulus-formplayer (this will auto-generate package-lock.json)
cd ../../formulus-formplayer
npm install

# Start the development server
npm start
```

The `package-lock.json` will be automatically generated/updated when you run `npm install` in the `formulus-formplayer` directory.

## Benefits

- **Single source of truth** - Tokens are now managed in one place (`packages/tokens`)
- **Automatic sync** - Changes to tokens automatically propagate to formplayer
- **No duplication** - Removed duplicate code

---

**Note**: I tested this works and the development server started successfully showing the use of token for the UI
 
![new](https://github.com/user-attachments/assets/db8f8162-c313-4202-9e28-2e7d3c4763ec)

Bahati308 commented:
https://github.com/user-attachments/assets/6007c4da-0956-45f0-bcc8-a23ecef95ba1

this is super dope, @IamLRBA, even Formulus can testify

IamLRBA reviewed (approved):
I see the tokens are being reflected in Formulus as well
Nice going @Bahati308
